src/components/TabularView.vue

```diff
@@ -0,0 +1,371 @@
+
+<script setup>
```

Comment:
This is the most complex component. I wanted to chop it into multiple components but did not have the time.